content/en/apps/guides/hosting/core-developer.md

```diff
@@ -0,0 +1,235 @@
+---
+title: "CHT Core Developer Hosting"
+linkTitle: "CHT Core Developer Hosting"
+weight: 40
+description: >
+  Compile the CHT Core from source when doing Core development
+---
+
+{{% alert title="Note" %}} This guide assumes you are a CHT Core developer wanting to compile the CHT Core from source code to make commits to the [public GitHub repository](https://github.com/medic/cht-core). To set up a hosting environment for devoloping apps, see the [app guide]({{< relref "apps/guides/hosting/app-developer.md" >}}).
+
+To deploy the CHT in production, see either [AWS hosting]({{< relref "apps/guides/hosting/self-hosting.md" >}}) or [Self hosting]({{< relref "apps/guides/hosting/ec2-setup-guide.md" >}}){{% /alert %}}
+
+
+## The Happy Path Installation
+
+This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation. These instructions should work verbatim on Ubuntu 18-22, but will need tweaks for MacOS (via `brew`) or Windows (via WSL2).
+
+### Install NodeJS, npm, grunt and Docker
+
+First, update your current packages:
+
+```shell
+sudo apt update&&sudo apt -y dist-upgrade
```

Comment:
Is the "dist-" part necessary? I'd have just expected a `sudo apt upgrade`

Reply:
Based on my reading [here](https://linoxide.com/difference-between-apt-get-upgrade-and-dist-upgrade/) and [here](https://askubuntu.com/a/226213), it is now a best practice for me for all but production instances (of which I run none b/c personal servers are not "production" ;) . I think it fits here in dev land, so resolving, but happy to get some push back!